glass/fields.py

```diff
@@ -402,6 +410,8 @@ def _generate_grf(
         if j is not None:
             y[:, j] = z
 
+        alm = np.asanyarray(glass.glass_to_healpix_spectra(alm))  # type: ignore[arg-type]
```

Comment:
IIRC, the `alm`s should be converted back to healpix ordering before being passed into `hp` functions. I have used the `glass.glass_to_healpix_spectra` because it does the ordering in the same way.

Reply:
Thanks! Should I just revert #527 so that we don't use functions named for `spectra` inside functions meant for `alm`?

Reply:
Sounds good! I will wrap that up in another PR, which can be merged before this one to make everything cleaner.